src/pages/cardsPage/CardsPage.jsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState, useEffect } from 'react';
+import { Box, TextField } from '@mui/material';
+import { UserCard } from '../../components/UserCard/UserCard';
+import './CardsPageStyles.css';
+import { fetchUserData } from '../../api/usersApi';
+import { useLocalStorage } from '../../hooks/useLocalStorage';
+
+function CardsPage() {
+  const [users, setUsers] = useState([]);
+  const [favorites, toggleFavorite] = useLocalStorage([]);
+
+  useEffect(() => {
+    (async () => {
+      const userData = await fetchUserData();
+      const dataSource = users.length === 0? userData: users;
+      setUsers(dataSource.map(user => ({ ...user, isFavorite: checkIfFavorite(user.id) })));
+    })();
+  }, [favorites]);
+
+  const checkIfFavorite = (id) => {
+    const favorite = favorites.find(user => user.id === id);
+    return favorites.find(user => user.id === id) ? favorite.isFavorite : false;
```

Comment:
Another way of solving this 
```suggestion
    return users
             .map(user => { 
                if (user.id === id) 
                  return user.isFav 
                })
              .some(user => user === true)
```